content/docs/rfcs/38/README.md

```diff
@@ -0,0 +1,652 @@
+
+
+The node then participates in a number of query rounds in which it
+solicits other node's opinion in query rounds.  Given a set of `N`
+leaderless computational nodes, a gossip-based protocol is presumed to
+exist which allows members to discover, join, and leave a weakly
+transitory maximally connected graph.  Joining this graph allows each
+node to view a possibly incomplete node membership list of all other
+nodes.  This view may change as the protocol advances, as nodes join
+and leave.  Under generalized Internet conditions, the membership of
+the graph would experience a churn rate varying across different
+time-scales, as the protocol rounds progress.  As such, a given node
+may not have a view on the complete members participating in the
+consensus on a proposal in a given round.
+
+The algorithm is divided into 4 phases:
+1. Querying
+2. Computing `confidence`, `evidence`, and `accumulated evidence`
+3. Transition function
+4. Opinion and Decision
+
+<!-- NOTE from CP: not sure this fits, commenting for now -->
+<!-- ### Proposal Identification
+
+The node has a semantics and serialization of the proposal, of which
+it sets an initial opinion:
+
+     opinion
+        <-- initial opinion on truth of the proposal
+            as one of: {NO, NONE, YES}
+    
+The proposal proceeds in asynchronous rounds, in which each node
+queries `k` randomly sampled nodes for their opinions until a decision
+about local finality is achieved.  The liveness of the algorithm is
+severely constrained in the absence of timeouts for a round to
+proceed.  When a given node has finalized its decision on the
+proposal, it enters a quiescent state in which it optionally discards
+all information gathered during the query process retaining only the
+final opinion on the truth of the proposal. -->
+
+### Setup Parameters
+
+The node initializes the following integer ratios as constants:
+```
+# The following values are constants chosen with justification from experiments
+# performed with the adversarial models
+
+# 
+confidence_threshold
+  <-- 1   
+         
+# constant look ahead for number of rounds we expect to finalize a
+# decision.  Could be set dependent on number of nodes 
+# visible in the current gossip graph.
+look_ahead 
+  <-- 19
+
+# the confidence weighting parameter (aka alpha_1)
+certainty 
+  <-- 4 / 5  
+doubt ;; the lack of confidence weighting parameter (aka alpha_2)
+  <-- 2 / 5 
+
+k_multiplier     ;; neighbor threshold multiplier
+  <-- 2
+
+;;; maximal threshold multiplier, i.e. we will never exceed 
+;;; questioning k_initial * k_multiplier ^ max_k_multiplier_power peers
+max_k_multiplier_power 
+  <-- 4
+    
+;;; Initial numbers of nodes queried in a round
+k_initial 
+  <-- 7
+
+;;; maximum query rounds before termination
+max_rounds
+   <-- 997 ;; TODO justify
```

Comment:
I'm not sure this is ok. In experimentations we considered a round **failed** if didn't reached consensus before round `100`.

Reply:
added note, can change later with some justification.